craft_application/services/lifecycle.py

```diff
@@ -265,3 +267,61 @@ def __repr__(self) -> str:
             f"{self.__class__.__name__}({self._app!r}, {self._project!r}, "
             f"{work_dir=}, {cache_dir=}, {build_for=}, **{self._manager_kwargs!r})"
         )
+
+    def _get_parallel_build_count(self) -> int:  # noqa: PLR0912
+        """Get the number of parallel builds to run."""
+        parallel_build_count = None
+        effect_env_name = "PARALLEL BUILD COUNT"
+
+        # fixed parallel build count environment variable
+        for env_name in [
+            (self._app.name + "_PARALLEL_BUILD_COUNT").upper(),
+            "CRAFT_PARALLEL_BUILD_COUNT",
+        ]:
+            if os.environ.get(env_name):
+                try:
+                    parallel_build_count = int(os.environ[env_name])
```

Comment:
Do you mind adding a debug statement to inform the user how the value was determined?

Like `emit.debug(f"Using parallel build count of {parallel_build_count} from environment variable {env_name!r}")`?

And probably two more debug statements later on where `parallel_build_count` is set.